### PR TITLE
feat: implemented functionality to add a profile picture by saving to…

### PR DESCRIPTION
**Pull request by Maiqel1**
# Description

This PR addresses the issues with fetching and updating the user's profile image from the backend rather than a third party service. It includes enhancements to ensure the profile image URL is correctly fetched and uploaded using the base API URL.

# Changes proposed

- Update the profile image fetching mechanism to use the backend API instead of the session.
- Ensure the image upload API correctly handles the base URL.

**LINEAR TICKET**
https://linear.app/nnb/issue/NNB-18/implement-change-user-profile-picture-boilerplate

## **What did you do?**

- Modified the API calls to fetch the profile image from the backend using the base URL.
- Implemented error handling for unauthorized requests during image upload and profile data fetching.
- Updated the image upload process to ensure the correct URL is used.

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.
- [x] I am making a pull request against the **dev branch** (left side).
- [x] My commit messages styles matches our requested structure.
- [x] My code additions will fail neither code linting checks nor unit test.
- [x] I am only making changes to files I was requested to.

# Screenshots/Videos

https://github.com/user-attachments/assets/f613349a-7356-45da-b1f3-5471bb2fb1bc

### CODE REVIEW

File: src/app/dashboard/(admin)/admin/(settings)/settings/page.tsx [hunk 1]
```diff
@@ -133,20 +134,35 @@ export default function SettingsPage() {
...
+          body: formData,
+          headers: {
+            Authorization: `Bearer ${data?.access_token}`,
+          },
+        });
+
+        const uploadData = await uploadResponse.json();
+        if (uploadResponse.ok) {
+          const baseUrl = await getApiUrl();
+          const profilePicUrl = uploadData.data.profile_picture_url.replace('http://localhost:3009', baseUrl);
```
Comment (phoenixdahdev) [inline src/app/dashboard/(admin)/admin/(settings)/settings/page.tsx]: Why 😭😭

File: src/config/auth.config.ts
Comment (phoenixdahdev) [inline src/config/auth.config.ts:1]: no brother nobody is touching this

all teams are to use avatar_url

File: src/lib/auth.ts
Comment (phoenixdahdev) [inline src/lib/auth.ts:1]: very very very wrong

File: src/types/index.d.ts
Comment (phoenixdahdev) [inline src/types/index.d.ts:1]: nope

**Review by phoenixdahdev — CHANGES_REQUESTED** [review]

File: src/app/dashboard/(admin)/admin/(settings)/settings/page.tsx [hunk 2]
```diff
@@ -133,20 +134,35 @@ export default function SettingsPage() {
...
-        const data: CloudinaryAsset = await response.json();
-        payload.profile_picture = data.url;
-      });
+
+      if (image) {
+        const formData = new FormData();
+        formData.append("file", image!);
+
+        const baseUrl = await getApiUrl();
+        const UPLOAD_API_URL = `${baseUrl}/api/v1/profile/upload-image`;
```
Comment (phoenixdahdev) [inline src/app/dashboard/(admin)/admin/(settings)/settings/page.tsx]: concerning this which team are you we need to talk about this

**Review by phoenixdahdev — CHANGES_REQUESTED** [review]
also fix you lint issues

File: src/app/dashboard/(admin)/admin/(settings)/settings/page.tsx [hunk 3]
```diff
@@ -158,41 +190,45 @@ export default function SettingsPage() {
...
+        image: payload.profile_pic_url,
+      };
+
+      const event = new CustomEvent("session", {
+        detail: { session: { ...data, user: updatedUser } },
+      });
+
+      window.dispatchEvent(event);
+
+      await signIn("credentials", { redirect: false });
```
Comment (phoenixdahdev) [inline src/app/dashboard/(admin)/admin/(settings)/settings/page.tsx]: Sign-in again how

File: src/app/dashboard/(admin)/admin/(settings)/settings/page.tsx [hunk 4]
```diff
@@ -1,7 +1,7 @@
 "use client";
 
 import axios from "axios";
-import { useSession } from "next-auth/react";
+import { signIn, useSession } from "next-auth/react";
```
Comment (phoenixdahdev) [inline src/app/dashboard/(admin)/admin/(settings)/settings/page.tsx]: use the unstable_update

now imagine it is a google user how will you authenticate the user

File: src/app/dashboard/(admin)/admin/(settings)/settings/page.tsx (hunk 3 above)
Comment (phoenixdahdev) [inline src/app/dashboard/(admin)/admin/(settings)/settings/page.tsx]: credential sign bro

**Review by phoenixdahdev — CHANGES_REQUESTED** [review]

**Review by prudentbird — DISMISSED** [review]

**Comment by prudentbird** [comment]
Use avatar_url instead of profile pic URL

**Review by phoenixdahdev — DISMISSED** [review]

**Review by chumex412 — DISMISSED** [review]

**Comment by Maiqel1** [comment]
one last time, i had lint errors @incredible-phoenix246 😭😭😭😭😭😭

**Review by chumex412 — APPROVED** [review]

**Review by oreoluwa212 — APPROVED** [review]

**Pull request merged**